coalib/results/SourceRange.py

```diff
@@ -120,6 +120,39 @@ def expand(self, file_contents):
                                        tr.end.line,
                                        tr.end.column)
 
+    @enforce_signature
+    def affected_source(self, file_dict: dict):
+        r"""
+        Make an iterator that tells which are the lines affected in the
+        file within a specified range.
+
+        >>> from os.path import abspath
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=2)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n')})
+        ('    x = 2  \n',)
+
+        If more than one line is affected.
+
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=3)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n', '    print(x)  \n')})
+        ('    x = 2  \n', '    print(x)  \n')
+
```

Comment:
I'd add a sentence and example here:

    If the file indicated at the source range is not in the `file_dict` or the lines are not given, this will return `None`:

    >>> ...